src/tsdb.c

```diff
@@ -206,6 +206,37 @@ void RenameSeriesFrom(RedisModuleCtx *ctx, RedisModuleString *key) {
     renameFromKey = key;
 }
 
+void RestoreKey(RedisModuleCtx *ctx, RedisModuleString *keyname) {
+    Series *series;
+    RedisModuleKey *key = NULL;
+    if (SilentGetSeries(ctx, keyname, &key, &series, REDISMODULE_READ) != TRUE) {
+        return;
+    }
+
+    CompactionRule *rule = series->rules;
+    while (rule != NULL) {
+        RedisModuleKey *destKey = NULL;
+        Series *destSeries;
+        const int status =
+            SilentGetSeries(ctx, rule->destKey, &destKey, &destSeries, REDISMODULE_WRITE);
+        if (status != TRUE) {
+            goto next;
+        }
+
+        RedisModule_RetainString(ctx, keyname);
+        destSeries->srcKey = keyname;
+        if (destKey) {
+            RedisModule_CloseKey(destKey);
+        }
+next:
+        rule = rule->nextRule;
```

Comment:
Why do you need goto?

```suggestion
        if (status == TRUE) {
            RedisModule_RetainString(ctx, keyname);
            destSeries->srcKey = keyname;
            if (destKey) {
               RedisModule_CloseKey(destKey);
            }
        }
        rule = rule->nextRule;
```

Reply:
I agree, no reason for the goto.